fw/tls.c

```diff
@@ -820,6 +822,22 @@ tfw_tls_sni(TlsCtx *ctx, const unsigned char *data, size_t len)
 				 PR_TFW_STR(&srv_name));
 			return -ENOENT;
 		}
+
+		/* Save server name for later validation with authority from requests */
```

Comment:
In the code above, if we have `vhost != NULL`, then the SNI strictly matches the host name or it is SAN and its suffix matches the host name. So why it's not enough just to reference `vhost` and maybe introduce a flag whether we had SAN request?

TBD if our configuration allows any SNI (e.g. we matched the default vhost), then should we enforce the SNI validation against HTTP authority?

Reply:
>So why it's not enough just to reference vhost and maybe introduce a flag whether we had SAN request?

Then we will need to validate authority against certificate CN/SAN for each subsequent request in that connection, which is quite ugly and more resource-hungry than just string comparison.

Also we could make a configuration option to prevent [domain fronting attack](https://www.comparitech.com/blog/information-security/what-is-a-domain-fronting-attack/) or accept it as we currently do.